### ISSUE TITLE
</opml> has no method 'charAt'

### ISSUE THREAD

**Issue by nickdesaulniers** [isaacs/sax-js#146]
Was trying to convert [a list of streams from TuneIn](http://opml.radiotime.com/Search.ashx?query=aa2) from XML to JSON real quick.

```
var parser = require('sax').parser(true);
parser.onopentag = function (node) {
  console.log(node);
};
parser.write(require('fs').readFileSync('./songs.xml'), { encoding: 'utf8' }).close();
```

Gives the stack trace:
```
</opml> has no method 'charAt'
    at Object.write (/Users/Nicholas/mozilla/tunein/node_modules/sax/lib/sax.js:918:31)
    at Object.<anonymous> (/Users/Nicholas/mozilla/tunein/xml2json.js:7:8)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
```
using sax v0.6.1.

**Comment by nickdesaulniers** [comment]
Ah, probably looking for [this](https://github.com/danmactough/node-opmlparser) anyways.

**Comment by isaacs** [comment]
Yeah, it doesn't handle buffers properly

Workaround:

```javascript
var parser = require('sax').parser(true);
parser.onopentag = function (node) {
  console.log(node);
};

// note the paren moved on this line, so we send a string to the parser
parser.write(require('fs').readFileSync('./songs.xml', { encoding: 'utf8' })).close();
```

**Comment by isaacs** [comment]
Fixed on v1.1.6
